### PR TITLE
Adventure events initial implementation

### PR DESCRIPTION
**Pull request by jjayers99**
Added randomly generated draft events to town inns - available in roughly half of towns at any given time, draft pools heavily weighted toward more recent sets.

### CODE REVIEW

**Comment by kevlahnota** [comment]
```
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/RewardActor.java:35:8: Unused import - forge.assets.FImageComplex. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/RewardActor.java:50:8: Unused import - org.apache.commons.lang3.tuple.Pair. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/RewardActor.java:52:8: Unused import - java.awt.image.BufferedImage. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/AdventureEventController.java:3:8: Unused import - com.badlogic.gdx.files.FileHandle. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/AdventureEventController.java:5:8: Unused import - com.badlogic.gdx.utils.Json. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/CardUtil.java:11:8: Unused import - forge.adventure.data.ItemData. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/CardUtil.java:20:8: Unused import - forge.gamemodes.quest.QuestController. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/CardUtil.java:22:8: Unused import - forge.item.InventoryItem. [UnusedImports]
Error:  /home/runner/work/forge/forge/forge-gui-mobile/src/forge/adventure/util/CardUtil.java:28:8: Unused import - forge.util.MyRandom. [UnusedImports]
```

File: forge-gui-mobile/src/forge/adventure/scene/DuelScene.java [hunk 1]
```diff
@@ -336,14 +360,23 @@ public void enter() {
...
+            rules.setGamesPerMatch(1);
+        }
         rules.setPlayForAnte(false);
         rules.setMatchAnteRarity(true);
-        rules.setGamesPerMatch(1);
         rules.setManaBurn(false);
         rules.setWarnAboutAICards(false);
 
-        hostedMatch.setEndGameHook(() -> DuelScene.this.GameEnd());
+        //hostedMatch.setEndGameHook(() -> DuelScene.this.GameEnd());
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/scene/DuelScene.java:379]: What will happen if adventurewinlose is skipped via user settings? does the gameend is called somewhere else?

File: forge-gui-mobile/src/forge/screens/match/winlose/AdventureWinLose.java [hunk 2]
```diff
@@ -11,16 +11,26 @@ public class AdventureWinLose extends ControlWinLose {
...
-        v.getBtnContinue().setVisible(false);
+
+        if (lastGame.isMatchOver()) {
+            v.getBtnQuit().setText(Forge.getLocalizer().getMessage("lblBackToAdventure"));
+            //v.getBtnContinue().setVisible(false);
+        }
+        else{
+            v.getBtnContinue().setVisible(true);
+            v.getBtnContinue().setEnabled(true);
+            v.getBtnContinue().setText("CONTINUE MATCH");
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/screens/match/winlose/AdventureWinLose.java:22]: needs localizer translation

File: forge-gui-mobile/src/forge/adventure/scene/DuelScene.java [hunk 3]
```diff
@@ -202,10 +210,17 @@ public void setDungeonEffect(EffectData E) {
     public void enter() {
         GameHUD.getInstance().unloadAudio();
         Set<GameType> appliedVariants = new HashSet<>();
-        appliedVariants.add(GameType.Constructed);
+        if (eventData!= null && eventData.eventRules != null){
+            appliedVariants.add(eventData.eventRules.gameType);
+        }
+        else{
+            appliedVariants.add(GameType.Adventure);
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/scene/DuelScene.java:217]: the status scene should be updated and the achievements for the adventure gametype - achievementcollection

File: forge-gui-mobile/src/forge/adventure/scene/InnScene.java [hunk 4]
```diff
@@ -39,13 +59,17 @@ private InnScene() {
...
+
+        ui.onButtonPress("event", InnScene.this::startEvent);
     }
 
 
 
     public void done() {
-        GameHUD.getInstance().getTouchpad().setVisible(false);
-        Forge.switchToLast();
+        Forge.switchScene(lastGameScene==null?GameScene.instance():lastGameScene);
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/scene/InnScene.java:72]: why remove this: GameHUD.getInstance().getTouchpad().setVisible(false); does the innscene don't display the touchpad on android anymore when this is called?

File: forge-gui-mobile/src/forge/adventure/util/RewardActor.java [hunk 5]
```diff
@@ -27,23 +27,29 @@
...
+import forge.item.SealedProduct;
 import forge.sound.SoundEffectType;
 import forge.sound.SoundSystem;
 import forge.util.ImageFetcher;
 import forge.util.ImageUtil;
+import forge.util.MyRandom;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
 
+import java.awt.image.BufferedImage;
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/util/RewardActor.java:52]: buffered image on mobile? this isnt supported

File: forge-gui-mobile/src/forge/adventure/util/RewardActor.java [hunk 6]
```diff
@@ -304,6 +310,45 @@ public RewardActor(Reward reward, boolean flippable, RewardScene.Type type) {
...
+                        imageKey = ImageKeys.BOOSTER_PREFIX + editionCode + ((1 >= maxIdx) ? "" : ("_" + artIndex));
+                    }
+                }
+                catch (Exception e){
+                    //Comment did not contain the edition code, this is not a basic booster pack
+                }
+
+
+
+                //FImageComplex cardArt = CardRenderer.getCardArt(imageKey, false, false, false, false, false,false,false,false,false,false);
```
Comment (kevlahnota) [inline forge-gui-mobile/src/forge/adventure/util/RewardActor.java:342]: this should be generated and drawn via buffers like the tooltip and should be added to dispose method

File: forge-gui-mobile/src/forge/adventure/scene/InnScene.java (hunk 4 above)
Comment (jjayers99) [reply forge-gui-mobile/src/forge/adventure/scene/InnScene.java:72]: Unintentional removal during replacement of the next line. Will put this back.

File: forge-gui-mobile/src/forge/adventure/scene/DuelScene.java (hunk 3 above)
Comment (jjayers99) [reply forge-gui-mobile/src/forge/adventure/scene/DuelScene.java:217]: Good point. Will implement & test these this afternoon.

**Comment by Agetian** [comment]
There are some unused imports at the moment. Will need some correction.

Thanks for seeing this project through to this stage, very exciting! :)

**Comment by kevlahnota** [comment]
```
Error:  /home/runner/work/forge/forge/forge-gui/src/main/java/forge/localinstance/achievements/AdventureAchievements.java:3:8: Unused import - forge.game.GameType. [UnusedImports]
```

**Review by kevlahnota — APPROVED** [review]

**Pull request merged**